src/vit-servicing-station/vit-servicing-station-tests/src/common/data/generator/arbitrary/raw_snapshot.rs

```diff
@@ -0,0 +1,66 @@
+use std::iter;
+
+use chain_impl_mockchain::testing::TestGen;
+use snapshot_lib::{
+    registration::{Delegations, Delegations::*, VotingRegistration},
+    RawSnapshot,
+};
+
+use super::ArbitraryGenerator;
+
+const LEGACY_DELEGATION: usize = 1;
+const NEW_DELEGATION: usize = 0;
+
+#[derive(Clone)]
+pub struct ArbitraryRawSnapshotGenerator {
+    id_generator: ArbitraryGenerator,
+}
+
+impl Default for ArbitraryRawSnapshotGenerator {
+    fn default() -> Self {
+        Self {
+            id_generator: ArbitraryGenerator::new(),
+        }
+    }
+}
+
+impl ArbitraryRawSnapshotGenerator {
+    pub fn legacy_delegation(&mut self) -> Delegations {
+        Delegations::Legacy(TestGen::identifier().into())
+    }
+
+    pub fn new_delegation(&mut self) -> Delegations {
+        let size = self.id_generator.random_size();
+        Delegations::New(
+            iter::from_fn(|| Some((TestGen::identifier().into(), self.id_generator.next_u32())))
+                .take(size)
+                .collect(),
+        )
+    }
+
+    pub fn delegation(&mut self) -> Delegations {
+        let delegation_type = self.id_generator.random_index(1);
+        match delegation_type {
+            LEGACY_DELEGATION => self.legacy_delegation(),
```

Comment:
Why not just use 0 or 1 instead of consts? Currently it suggests that those consts are some macros or global settings while it just for choosing which delegation type to generate. I would just simply wrote:

```
        match self.id_generator.next_u32() % 2 {
            0 => self.legacy_delegation(),
            1 = > self.new_delegation(),
        }
``` 

And then we can get rid of those macros

Reply:
yeah you are right, I wanted somehow to make it more clear but it doesnt really need to